src/realm/binary_data.hpp

```diff
@@ -57,6 +57,10 @@ class BinaryData {
     template <class T, class A>
     explicit BinaryData(const std::basic_string<char, T, A>&);
 
+    // BinaryData does not store data, callers must manage their own strings.
+    template <class T, class A>
+    BinaryData(const std::basic_string<char, T, A>&&) = delete;
```

Comment:
One situation is a function returning `const T`, which otherwise be handled by the `const &` overload and result in a `BinaryData` referencing data that is owned by a temporary.